src/peep/peep.c

```diff
@@ -2420,6 +2423,9 @@ void peep_update_ride_sub_state_7(rct_peep* peep){
 			}
 
 			ride_entry = get_ride_entry(ride->subtype);
+			if (ride_entry == NULL) {
+				return;
+			}
```

Comment:
It might be better to instead skip the code that uses ride entry and vehicle entry. Let the peep still walk to the exit.